frame/nfts/src/common_functions.rs

```diff
@@ -30,6 +32,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Collection::<T, I>::get(collection).map(|i| i.owner)
 	}
 
+	/// Convert signer into account id.
+	pub fn signer_to_account(signer: MultiSigner) -> Result<T::AccountId, DispatchError> {
+		Ok(T::AccountId::decode(&mut signer.into_account().as_ref())
```

Comment:
```suggestion
		Ok(T::AccountId::decode_all(&mut signer.into_account().as_ref())
```

Here you are decoding an `AccountId` from a `AccountId32`. That is not guaranteed to work.  
I think instead your pallet should have a `Lookup` or `Convert` config item which can convert `AccountId32` to `AccountId` or you mark your `System::AccountId` as `From<AccountId32>`.